metagraph/src/cli/load/load_annotated_graph.cpp

```diff
@@ -23,15 +23,13 @@ using mtg::common::logger;
 
 std::unique_ptr<AnnotatedDBG> initialize_annotated_dbg(std::shared_ptr<DeBruijnGraph> graph,
                                                        const Config &config) {
+    uint64_t max_index = graph->get_base_graph().max_index();
+
     if (graph->get_mode() == DeBruijnGraph::PRIMARY) {
         graph = std::make_shared<CanonicalDBG>(graph);
         logger->trace("Primary graph was wrapped into canonical");
     }
```

Comment:
There too many degrees of freedom. Better make the logic as simple as possible while still supporting all features. Choose whether one should pass an unwrapped graph (and hence, no `get_base_graph()` is needed to be called) or never a primary graph (so it must always be wrapped when passed, and no wrapping is needed).

The code will be simpler and easier to maintain/use.

Reply:
ok, makes sense. I added an assert here to make sure only a base graph is passed